crimemapper/graph_calcs.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+
+"""Graph calculations for graph view."""
+from crimemapper.models import (
+    DBSession,
+    Entry,
+)
+from collections import Counter
+import random
+
+
+UPPER_DICT = {
+    "Assault": ["ASSAULT"],
+    "Theft": [
+        "BIKE THEFT", "BURGLARY", "CAR PROWL", "MAIL THEFT", "PICKPOCKET",
+        "PURSE SNATCH", "SHOPLIFTING", "THEFT OF SERVICE", "VEHICLE THEFT",
+        "BURGLARY-SECURE PARKING-RES"
+    ],
+    "Sin": ["GAMBLE", "PORNOGRAPHY", "PROSTITUTION"],
+    "Drugs": [
+        "DUI", "LIQUOR VIOLATIONS", "NARCOTICS", "STAY OUT OF AREA OF DRUGS"
+    ],
+    "Fraud": ["COUNTERFEIT", "EMBEZZLE", "FORGERY", "FRAUD"],
+    "Homicide": ["HOMICIDE"],
+    "Property": [
+        "ILLEGAL DUMPING", "LOST PROPERTY", "OTHER PROPERTY",
+        "PROPERTY DAMAGE", "RECKLESS BURNING", "RECOVERED PROPERTY",
+        "STOLEN PROPERTY",
+    ],
+    "Robbery": ["ROBBERY"],
+    "Weapons": ["WEAPONS"],
+    "Disturbances": [
+        "DISORDERLY CONDUCT", "DISPUTE", "FIREWORK", "PUBLIC NUISANCE",
+        "THREATS", "TRESPASS"
+    ],
+    "Other": [
+        "ANIMAL COMPLAINT", "BIAS INCIDENT", "ELUDING", "ESCAPE", "EXTORTION",
+        "HARBOR CALLS", "INJURY", "LOITERING", "METRO", "OBSTRUCT", "TRAFFIC",
+        "WARRANT ARREST", "VIOLATION OF COURT ORDER"
+    ],
+}
+
+
+CATEGORY_MAPPING = {tuple(val): key for key, val in UPPER_DICT.items()}
+
+
+def get_category(word):
+    """Invert keys and items."""
+    for words in CATEGORY_MAPPING:
+        if word in words:
+            return CATEGORY_MAPPING[words]
+    else:
+        return None
+
+
+MAIN_RESULTS = {}
+
+
+def main_db_call():
+    """Cache a db call to close the session."""
+    if 'already_called' not in MAIN_RESULTS:
+        results = DBSession().query(
+            Entry.summarized_offense_description
+        ).all()
+        MAIN_RESULTS['already_called'] = results
+        try:
+            MAIN_RESULTS['already_called'][0]
+        except IndexError:
+            raise ImportError
```

Comment:
This smells funny to me.  Is `ImportError` _really_ the right thing to raise here?